pkg/scalers/azure_servicebus_scaler.go

```diff
@@ -249,6 +262,7 @@ func (s *azureServiceBusScaler) getServiceBusNamespace() (*servicebus.Namespace,
 		namespace.Name = s.metadata.namespace
 	}
 
+	namespace.Suffix = s.metadata.endpointSuffix
```

Comment:
I think this is all it takes for the servicebus client to work in other cloud environments, hope I didn't miss anything.

Reply:
@ahmelsayed Can you confirm this?

Reply:
sorry for the delay, I've been out for a while and just got back. 

There is also `namespace.Environment` that I see in the API. I don't have an easy way to validate it, but I'll try to get an environment where I can.

Reply:
From what I could tell, `namespace.Environment` is only used for getting the service bus resource ID when calling [NamespaceWithEnvironmentBinding](https://github.com/Azure/azure-service-bus-go/blob/master/namespace.go#L146) to configure a namespace using environment details. Since we either create the namespace with connection string or with pod identity, I think `namespace.Environment` isn't required. That also makes sense to me because as far as I know the service bus client should also support cloud environments that are not the [publicly known clouds](https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go#L34), like air gapped clouds.

@ahmelsayed I might be able to test it in one of our clusters in US Gov if that helps.